quickwit/quickwit-config/src/node_config/serialize.rs

```diff
@@ -810,38 +870,74 @@ mod tests {
             .build_and_validate(&HashMap::new())
             .await
             .unwrap();
-            assert_eq!(node_config.rest_listen_addr.to_string(), "127.0.0.1:7280");
+            assert_eq!(
+                node_config.rest_config.listen_addr.to_string(),
+                "127.0.0.1:7280"
+            );
             assert_eq!(node_config.gossip_listen_addr.to_string(), "127.0.0.1:7280");
             assert_eq!(node_config.grpc_listen_addr.to_string(), "127.0.0.1:7281");
         }
         {
             let node_config = NodeConfigBuilder {
                 listen_address: default_listen_address(),
-                rest_listen_port: ConfigValue::for_test(1789),
+                rest_config_builder: RestConfigBuilder {
+                    listen_port: Some(1789),
+                    ..Default::default()
+                },
                 ..Default::default()
             }
             .build_and_validate(&HashMap::new())
             .await
             .unwrap();
-            assert_eq!(node_config.rest_listen_addr.to_string(), "127.0.0.1:1789");
+            assert_eq!(
+                node_config.rest_config.listen_addr.to_string(),
+                "127.0.0.1:1789"
+            );
             assert_eq!(node_config.gossip_listen_addr.to_string(), "127.0.0.1:1789");
             assert_eq!(node_config.grpc_listen_addr.to_string(), "127.0.0.1:1790");
         }
         {
             let node_config = NodeConfigBuilder {
                 listen_address: default_listen_address(),
-                rest_listen_port: ConfigValue::for_test(1789),
                 gossip_listen_port: ConfigValue::for_test(1889),
                 grpc_listen_port: ConfigValue::for_test(1989),
+                rest_config_builder: RestConfigBuilder {
+                    listen_port: Some(1789),
+                    ..Default::default()
+                },
                 ..Default::default()
             }
             .build_and_validate(&HashMap::new())
             .await
             .unwrap();
-            assert_eq!(node_config.rest_listen_addr.to_string(), "127.0.0.1:1789");
+            assert_eq!(
+                node_config.rest_config.listen_addr.to_string(),
+                "127.0.0.1:1789"
+            );
             assert_eq!(node_config.gossip_listen_addr.to_string(), "127.0.0.1:1889");
             assert_eq!(node_config.grpc_listen_addr.to_string(), "127.0.0.1:1989");
         }
+        {
+            // Check that `rest_listen_port` can still be used for backward compatibility.
+            let node_config = NodeConfigBuilder {
```

Comment:
Nit: this probably should go in a dedicated test. This one is already quite lengthy.